analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -92,10 +92,47 @@ void AssignmentTransitivity()
         _ = b1.Value;                                        // Noncompliant
     }
 
-    void AssignmentAndDeconstruction()
+    void AssignmentAndDeconstructionUpcasting()
```

Comment:
These are nice UTs that will help with deconstruction implementation in the future.

Can you also add nested deconstruction? 
Like `(a, (b, (c, d), e))` direct 1:1, and `(a, (b, (c, d), e))` where `(c, d)` is result of an object deconstruction via custom reconstruction method.

Reply:
[DRAFT]
Added nested deconstruction with 2 and 3 levels, including a scenario with typed and untyped discard.

I couldn't find a way to deconstruct a `(int?, int?)` value tuple into `int`: the following doesn't seem to work (the deconstruction method is not picked up):
```cs
public static class Extensions
{
    public static void Deconstruct(this ValueTuple<int, int> tuple, out int v1, out int v2)
    {
        v1 = 42;
        v2 = 42;
    }
}
``` 

```cs
(int?, int?) couple = (null, null);
(int i1, int i2) = couple; // CS0266 Cannot implicitly convert type 'int?' to 'int' ...
```

Reply:
The case I meant was this - the number of variables on the left doesn't match number on the right due to the `aaa` and `bbb` deconstruction
```
public class Sample
{
    void WithNestedDeconstructor(Sample deconstructed)
    {
        (_, (int? a, (int? aaa, int? bbb), int? c)) = (42, (42, deconstructed, null));
        _ = a.Value; // Compliant
        _ = c.Value; // FN
    }

    public void Deconstruct(out int? aaa, out int? bbb)
    {
        aaa = null;
        bbb = null;
    }
}
```